### ISSUE TITLE
Miner / Validator Version Control

### ISSUE THREAD

**Issue by grantdfoster** [masa-finance/masa-bittensor#167]
## Description

We need to leverage some sort of version control to ensure that Validators and Miners are running the latest code, similar to how the Masa Protocol functions.  It was suggested to look into the Axon object itself, as metadata there gets reported to the metagraph, and it's possible we could enforce versioning from there.  Also worth asking around in the Bittensor Discord dev channels for ideas.

**Comment by grantdfoster** [comment]
The hyperparameter `weights_version` is used to set a specific validator version on the subnet, ensuring code consistency across validators.  I don't believe there is a similar mechanism for miners, still investigating.

**Comment by grantdfoster** [comment]
One aspect that is strange is that our current `weights_version` on the subnet is 0, whereas our validators are sending `10` in their version key, i.e. version `0.1.0`.  It seems weights are being set correctly despite the mismatch.  One thought is that when you set a weights_version of 0 on the subnet, it allows all of them.  Will test this theory on devnet where I control a subnet.

**Comment by grantdfoster** [comment]
I have verified that when validators are running a _spec_version__ that is less than what is defined in the `weights_version` hyperparameter, they are not able to set weights!  This is great.  Adding to the production docs, but it will be important that we set this value to 1 when we launch the subnet on production.  Similarly, I have made an edit to __version__ in the validator to be 0.0.1.  The logic is such that 1.0.0 == 100, 0.1.0 == 10, etc.
